silkworm/rpc/http/request_handler.cpp

```diff
@@ -158,21 +158,24 @@ Task<void> RequestHandler::handle_request(commands::RpcApiTable::HandleMethod ha
 }
 
 Task<void> RequestHandler::handle_request(commands::RpcApiTable::HandleStream handler, const nlohmann::json& request_json) {
-    try {
-        auto io_executor = co_await boost::asio::this_coro::executor;
+    auto io_executor = co_await boost::asio::this_coro::executor;
 
-        co_await channel_->open_stream();
-        ChunkWriter chunk_writer(*channel_);
-        json::Stream stream(io_executor, chunk_writer);
+    co_await channel_->open_stream();
+    ChunkWriter chunk_writer(*channel_);
+    json::Stream stream(io_executor, chunk_writer);
 
+    try {
         co_await (rpc_api_.*handler)(request_json, stream);
-
-        co_await stream.close();
     } catch (const std::exception& e) {
         SILK_ERROR << "exception: " << e.what();
+        const auto error = make_json_error(request_json, 100, e.what());
+        stream.write_json(error);
     } catch (...) {
         SILK_ERROR << "unexpected exception";
+        const auto error = make_json_error(request_json, 100, "unexpected exception");
+        stream.write_json(error);
     }
+    co_await stream.close();
```

Comment:
I suggest using two nested try-catch blocks here in order to have both the improved JSON errors you are adding and error handling for `open_stream` and `stream.close`

Reply:
Right, done